sql/analyzer/validation_rules_test.go

```diff
@@ -141,6 +140,8 @@ func TestValidateGroupByErr(t *testing.T) {
 		plan.NewResolvedTable(child, nil, nil),
 	)
 
+	err = sql.SystemVariables.SetGlobal("sql_mode", "STRICT_TRANS_TABLES,NO_ENGINE_SUBSTITUTION,ONLY_FULL_GROUP_BY")
```

Comment:
Should we go ahead and just make this the default for `sql_mode`? That would get us one step closer to matching MySQL's default `sql_mode`, so seems like a good change.

Reply:
Yeah this should be the default; even better since dolt already did this by default just without setting the variable.
For some reason, my MySQL server didn't have this option set, so I assumed the default was not strict.